server/src/main/java/org/opensearch/index/seqno/LocalCheckpointTracker.java

```diff
@@ -131,22 +131,49 @@ public synchronized void markSeqNoAsProcessed(final long seqNo) {
      * @param seqNo the sequence number to mark as persisted
      */
     public synchronized void markSeqNoAsPersisted(final long seqNo) {
-        markSeqNo(seqNo, persistedCheckpoint, persistedSeqNo);
+        markSeqNo(seqNo, persistedCheckpoint, persistedSeqNo, false);
     }
 
-    private void markSeqNo(final long seqNo, final AtomicLong checkPoint, final LongObjectHashMap<CountedBitSet> bitSetMap) {
+    /**
+     * Updates the processed checkpoint to the provided sequence number if segment replication is enabled
+     *
+     * @param seqNo the sequence number to mark as processed
+     */
+    public synchronized void fastForwardProcessedSeqNo(final long seqNo) {
+        markSeqNo(seqNo, processedCheckpoint, null, true);
+    }
+
+    private boolean shouldUpdateSeqNo(final long seqNo, boolean segrep, final AtomicLong checkpoint) {
```

Comment:
As @mch2 alluded to in an earlier comment - if you're having to write variables to model use-cases, then that's usually a sign of an overfit. Consider having this method signature be:

```
private boolean shouldUpdateSeqNo(final long seqNo, final AtomicLong lowerBound, @Nullable final AtomicLong upperBound)
```

Also see my other comment.

Reply:
Followed the signature suggested